### ISSUE TITLE
Cannot reproduce results on cascade hrnet

### ISSUE THREAD

**Issue by Mabinogiysk** [open-mmlab/mmdetection#2643]
Python: 3.7.7 (default, Mar 26 2020, 15:48:22) [GCC 7.3.0]
GCC: gcc (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
PyTorch: 1.3.1
TorchVision: 0.4.2
OpenCV: 4.2.0
MMCV: 0.4.3
MMDetection: 1.1.0+2452761
MMDetection Compiler: GCC 7.5
MMDetection CUDA Compiler: 10.0

-----
I have trained faster_rcnn_r50_fpn_1x、cascade_rcnn_hrnetv2p_w32_20e、cascade_rcnn_r50_fpn_1x on my dataset. Performance of cascade_rcnn_hrnetv2p_w32_20e is much higher than faster_rcnn_r50_fpn_1x but slightly lower than cascade_rcnn_r50_fpn_1x.
I have evaluated cascade_rcnn_hrnetv2p_w32_20e on coco dataset with pretrained model, everything seems fine. Then I trained cascade_rcnn_hrnetv2p_w32_20e from scrath on coco2017 dataset with 8 gpu:
```
sh ./tools/dist_train.sh ${config_file} 8 --work_dir ${OUTPUT_DIR}/ --autoscale-lr
```
AP of the model is 0.212.
Is any suggestions?

**Comment by ZwwWayne** [comment]
You do not need to use `--autoscale-lr` since the config already set the learning rate for 8 GPU setting. So it is very likely that your LR is too large.

**Comment by Mabinogiysk** [comment]
> You do not need to use `--autoscale-lr` since the config already set the learning rate for 8 GPU setting. So it is very likely that your LR is too large.

It may have some bugs in `--autoscale-lr`. I found that lr was always 0.00250, one eighth of the value I set in config.py(0.02) whether I am using 2 GPUs or 8 GPUs. Now I am training from scrath without `--autoscale-lr`, lr is 0.02. After 11 epoches, the loss is around 1.0, which I think is too large.

**Comment by ZwwWayne** [comment]
1. `training from scrath` is a little bit vague, usually it means you train a model without using pre-trained backbone, did you do that? Or you just simply rerun a config without `--autoscale-lr`?
2. It is quite normal for cascade methods have loss around 1.0, some logs can be found in our model zoo for your reference, e.g., [cascade_mask_rcnn_r50_fpn_1x](https://open-mmlab.s3.ap-northeast-2.amazonaws.com/mmdetection/v2.0/cascade_rcnn/cascade_mask_rcnn_r50_fpn_1x_coco/cascade_mask_rcnn_r50_fpn_1x_coco_20200203_170449.log.json). We will provide model models and logs in the model zoo.

**Comment by Mabinogiysk** [comment]
> 1. `training from scrath` is a little bit vague, usually it means you train a model without using pre-trained backbone, did you do that? Or you just simply rerun a config without `--autoscale-lr`?
> 2. It is quite normal for cascade methods have loss around 1.0, some logs can be found in our model zoo for your reference, e.g., [cascade_mask_rcnn_r50_fpn_1x](https://open-mmlab.s3.ap-northeast-2.amazonaws.com/mmdetection/v2.0/cascade_rcnn/cascade_mask_rcnn_r50_fpn_1x_coco/cascade_mask_rcnn_r50_fpn_1x_coco_20200203_170449.log.json). We will provide model models and logs in the model zoo.

I cannot train cascade_rcnn_hrnetv2p_w32_20e on my own dataset and I suspect it has something to do with training. So I trained it on coco dataset without using pre-trained backbone and tried to reproduce the results.
I have trained another `cascade hrnetv2p` on my own dataset without `--autoscale-lr` . Again, performance of the model slight lower than that of `cascade_rcnn_r50_fpn_1x`

**Comment by Mabinogiysk** [comment]
New training without `--autoscale-lr` failed again, AP is 0.25.

**Comment by Mabinogiysk** [comment]
Evaluation on coco2017:
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.255
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.401
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.270
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.128
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.265
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.362
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 0.257
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = 0.399
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.415
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.210
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.445
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.565

**Comment by ZwwWayne** [comment]
The performance reported in the model zoo is produced with ImageNet pre-trained models.
Training without using a pre-trained backbone usually obtain worse results with a 1x schedule and you can check details in [here](https://github.com/open-mmlab/mmdetection/tree/master/configs/scratch).

**Comment by Mabinogiysk** [comment]
> The performance reported in the model zoo is produced with ImageNet pre-trained models.
> Training without using a pre-trained backbone usually obtain worse results with a 1x schedule and you can check details in [here](https://github.com/open-mmlab/mmdetection/tree/master/configs/scratch).

Many thanks.
But how can I be sure that there is no problem in training process? Or is that normal that the performance of cascade_rcnn_hrnetv2p_w32 is slower than that of cascade_rcnn_r50_fpn_1x?

**Comment by ZwwWayne** [comment]
Usually, if you can see the training logs print out normally, it is OK.
About the training speed, cascade_rcnn_hrnetv2p_w32 should be slower than cascade_rcnn_r50_fpn.

**Comment by Mabinogiysk** [comment]
> Usually, if you can see the training logs print out normally, it is OK.
> About the training speed, cascade_rcnn_hrnetv2p_w32 should be slower than cascade_rcnn_r50_fpn.

Sorry for my unclear expression. What I mean is that the AP of cascade_rcnn_hrnetv2p_w32_20e is much higher than that of faster_rcnn_r50_fpn_1x but slightly lower than that of cascade_rcnn_r50_fpn_1x on my dataset. Now I am working on finding out the reason.

**Comment by ZwwWayne** [comment]
Hi @Mabinogiysk ,
If these configs on other datasets do not exhibit the same trends they have on the coco dataset, then there may be some bug or non-trivial things.

**Comment by Mabinogiysk** [comment]
Thanks

**Comment by Mabinogiysk** [comment]
I have trained faster_rcnn_hrnetv2p_w32_1x、cascade_rcnn_hrnetv2p_w32_20e and cascade_rcnn_r50_fpn_1x on my dataset. faster_rcnn_hrnetv2p_w32_1x and cascade_rcnn_r50_fpn_1x both work fine but the AP of cascade_rcnn_hrnetv2p_w32_20e is slightly lower. All model have trained with pre-trained model and the only difference in config.py is the path of `ann_file` and `img_prefix` in training process.
Now I will try it on mmdetection2.0.
